morf-core/src/test/java/org/alfasoftware/morf/upgrade/TestUpgradePathFinder.java

```diff
@@ -131,6 +133,42 @@ public void testExactPath() {
     assertSame("Third upgrades step", AddJamType.class, path.get(2).getClass());
   }
 
+  @Test(expected = IllegalStateException.class)
+  public void testfindDiscrepanciesThrowsException() {
+    // This test will check that an IllegalStateException is thrown as we are trying to run an upgrade step that has already run
+    List<Class<? extends UpgradeStep>> upgradeSteps = new ArrayList<>();
+    upgradeSteps.add(AddJamType.class);
+    upgradeSteps.add(AddCakeTable.class);
+    upgradeSteps.add(AddDiameter.class);
+    upgradeSteps.add(InsertAVictoriaSpongeRowUsingDSL.class);
+    UpgradePathFinder pathFinder = makeFinder(upgradeSteps, appliedSteps(AddJamType.class, AddCakeTable.class));
+
+    Map<String, String> upgradeAudit = new HashMap<>();
+    upgradeAudit.put(InsertAVictoriaSpongeRowUsingDSL.class.getName(), "0fde0d93-f57e-405c-81e9-245ef1ba0591");
+    upgradeAudit.put(AddDiameter.class.getName(), "6bb40ce0-2578-11e0-ac64-0800200c9a61");
+    try {
+      pathFinder.findDiscrepancies(upgradeAudit);
+    } catch (IllegalStateException e) {
+      assertTrue(e.getMessage().contains(InsertAVictoriaSpongeRowUsingDSL.class.getAnnotation(UUID.class).value()));
+      throw e;
+    }
+  }

```

Comment:
Try to use `assertThrows` in this test, it will simplify things a lot.
Also, `assertThrows` also returns the caught exception, so you can easily examine the message afterwards.